### PR TITLE
For #5133: Allow intent to bypass onboarding screen for perf

### PR DESCRIPTION
**Pull request by MarcLeclair**

### CODE REVIEW

File: app/src/main/java/org/mozilla/focus/perf/Performance.kt
Comment (MarcLeclair) [reply app/src/main/java/org/mozilla/focus/perf/Performance.kt]: I think that would be better in case it gets refactored / loss. Also, any change in one might be needed in the other. I think its just better for overall code health to have the duplicated code in one place

File: app/src/main/java/org/mozilla/focus/activity/MainActivity.kt
Comment (mcomella) [inline app/src/main/java/org/mozilla/focus/activity/MainActivity.kt:67]: nitpick (blocking): we need to update the kdoc on `ShowFirstRun` to say it's not just for tests.

Comment (mcomella) [inline app/src/main/java/org/mozilla/focus/activity/MainActivity.kt:67]: thought (non-blocking): this is just a summary of my understanding of this pattern. This is a timeline of the events:
```
appStore.dispatch(ShowFirstRun) // dispatches to appStore dispatcher thread
lifecycle.addObserver(navigator)

navigator.start() -> subscribe() // subscribes to appStore in onStart

navigator.subscribe$collect // changes screens on state change
```

One thing I don't understand is why the navigator is guaranteed to receive the event. Since the dispatch occurs on a background thread, could the state update occur before we subscribe in `onStart`? I'm not familiar with the flow API: maybe it's propagating the initial state to the subscribers and that's why it's safe? Or perhaps it's because the framework (probably) doesn't yield the UI thread until after `onStart` so we have time to subscribe before we collect?

@jonalmeida Do you know if this is safe?

---

A second thought is if there is anything in `HomeActivity.onCreate/onStop/etc.` that shouldn't be called because we've navigated to the onboarding screen. I don't think I see anything. It may be added in the future but I'm not sure how to protect against that.

**Review by mcomella — COMMENTED** [review]
Besides my open question about race conditions, this seems okay to me but I'd like to defer to jonalmeida's review.

btw, let's squash before we push.

**Review by mcomella — APPROVED** [review]
Apparently I requested changes so I have to approve to land this but I'd like to wait for jonalmeida's review before we land this.

File: app/src/main/java/org/mozilla/focus/activity/MainActivity.kt
Comment (jonalmeida) [reply app/src/main/java/org/mozilla/focus/activity/MainActivity.kt:69]: >  It may be added in the future but I'm not sure how to protect against that.

Does the intent that we send go directly to the `MainActivity` instead of the `IntentReceiverActivity` first? We could protect against this by processing the intent at the other activity and fire the action immediately before the `MainActivity` is even created.

Just a thought.

**Review by jonalmeida — APPROVED** [review]
I've left some comments for thought, but I don't think that should block us today if you need this to land for getting startup testing from intents to work.

File: app/src/main/java/org/mozilla/focus/perf/Performance.kt
Comment (MarcLeclair) [reply app/src/main/java/org/mozilla/focus/perf/Performance.kt:17]: Agreed, I think @mcomella  pointed that above. It is a duplicated class

**Pull request merged**
